Marlin/src/gcode/gcode.cpp

```diff
@@ -125,6 +130,43 @@ void GcodeSuite::dwell(millis_t time) {
   while (PENDING(millis(), time)) idle();
 }
 
+/**
+ * When G29_ADVANCED_RECOVERY_AND_RETRY is enabled, call G29() in
+ * a loop with recovery and retry handling.
+ */
+#if HAS_LEVELING && ENABLED(G29_ADVANCED_RECOVERY_AND_RETRY)
+
+void GcodeSuite::G29_with_retry() {
+    set_bed_leveling_enabled(false);
+    for (uint8_t i = 0; i < G29_MAX_RETRIES; i++) {
+      G29();
```

Comment:
Here's a problem. When you execute other G-code commands, their parameters replace the ones in the parser. So the original parameters to `G29` get lost, and if the last command to get executed before this call has parameters of its own, they will be sent to and interpreted by `G29`.